spring/boot-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ByteBufLeakTest.java

```diff
@@ -0,0 +1,154 @@
+                @Override
+                public NettyDataBuffer allocateBuffer(int initialCapacity) {
+                    final NettyDataBuffer buffer = super.allocateBuffer(initialCapacity);
+                    // Keep allocated buffers.
+                    allocatedBuffers.offer(buffer);
+                    return buffer;
+                }
+            };
+        }
+
+        @RestController
+        static class TestController {
+            @GetMapping("/mono")
+            Mono<String> mono() {
+                return Mono.just("hello, WebFlux!");
+            }
+
+            @GetMapping("/flux")
+            Flux<String> flux() {
+                return Flux.just("abc", "def", "ghi", "jkl", "mno");
+            }
+
+            @GetMapping("/empty")
+            Mono<String> empty() {
+                return Mono.empty();
+            }
+        }
+    }
+
+    @Rule
+    public TestRule globalTimeout = new DisableOnDebug(new Timeout(10, TimeUnit.SECONDS));
+
+    @LocalServerPort
+    int port;
+
+    @Test
+    public void confirmNoBufferLeak() throws Exception {
+        assert allocatedBuffers.isEmpty();
+        final HttpClient client = HttpClient.of("http://127.0.0.1:" + port);
+        for (int i = 0; i < 3; i++) {
+            assertThat(client.get("/mono").aggregate().join().contentUtf8())
+                    .isEqualTo("hello, WebFlux!");
+            assertThat(client.get("/flux").aggregate().join().contentUtf8())
+                    .isEqualTo("abcdefghijklmno");
+            assertThat(client.get("/empty").aggregate().join().contentUtf8())
+                    .isEmpty();
+        }
+
+        await().untilAsserted(this::makeSureReleasedAllAllocatedBuffers);
+    }
+
+    @Test
+    public void confirmNoBufferLeak_resetConnection() throws Exception {
+        assert allocatedBuffers.isEmpty();
+        for (int i = 0; i < 3 * 3; i++) {
+            try (Socket s = new Socket(NetUtil.LOCALHOST, port)) {
+                s.setSoLinger(true, 0);
+                final PrintWriter outWriter = new PrintWriter(s.getOutputStream(), false);
+                switch (i % 3) {
+                    case 0:
+                        outWriter.print("GET /mono HTTP/1.1\r\n\r\n");
+                        break;
+                    case 1:
+                        outWriter.print("GET /flux HTTP/1.1\r\n\r\n");
+                        break;
+                    case 2:
+                        outWriter.print("GET /empty HTTP/1.1\r\n\r\n");
+                        break;
+                }
+                outWriter.flush();
+            }
+        }
+
+        // The buffer allocation might be done after closing the socket. So waiting a little before
+        // checking the allocated buffers.
+        Thread.sleep(3000);
+
+        await().untilAsserted(this::makeSureReleasedAllAllocatedBuffers);
+    }
+
+    private void makeSureReleasedAllAllocatedBuffers() {
```

Comment:
- How about doing `await().untilAsserted(...)` inside this method, because this method is always used with `untilAsserted()`?
- How about renaming: `ensureAllBuffersAreReleased`?